tools/deployment/linux_check_init.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+#  Copyright (c) 2014-present, Facebook, Inc.
+#  All rights reserved.
+#
+#  This source code is licensed under the BSD-style license found in the
+#  LICENSE file in the root directory of this source tree. An additional grant
+#  of patent rights can be found in the PATENTS file in the same directory.
+
+SYSTEMD_SERVICE_DIR="/usr/lib/systemd/system"
+SYSTEMD_SYSCONFIG_DIR="/etc/sysconfig"
+SYSTEMD_SYSCONFIG_DIR_DEB="/etc/default"
+INITD_DIR="/etc/rc.d/init.d"
+INITD_DIR_DEB="/etc/init.d"
+
+# Check if OS is Debian based
+if ls /etc | grep -Fq debian_version; then
+  SYSTEMD_SYSCONFIG_DIR=$SYSTEMD_SYSCONFIG_DIR_DEB
+  INITD_DIR=$INITD_DIR_DEB
+fi
+
+# Check if system is using upstart
+if ps -p 1 | grep -Fq init && init --version | grep -Fq upstart; then
+  rm $SYSTEMD_SYSCONFIG_DIR/osqueryd
```

Comment:
Yes, that is possible. However, because of how Debian handles pre-uninstall scripts (namely, the fact that the pre-uninstall & post-uninstall scripts are run after both `dpkg --remove` and `dpkg --purge`), we could end up with even more warning/error messages on the Debian side. I'll test this later to be sure.

Right now, I'm looking for some theoretical command to manually register/de-register files from the package database. I'll have an update later today.